aggregator_core/src/datastore/tests.rs

```diff
@@ -99,6 +106,38 @@ async fn down_migrations(
     ephemeral_datastore.downgrade(0).await;
 }
 
+#[tokio::test]
+async fn retry_limit() {
+    install_test_trace_subscriber();
+    for max_transaction_retries in [0, 1, 1000] {
+        let ephemeral_datastore = EphemeralDatastoreBuilder::new().build().await;
+        let datastore = ephemeral_datastore
+            .datastore_with_max_transaction_retries(MockClock::default(), max_transaction_retries)
+            .await;
+
+        // The number of times the transaction was actually run.
+        let num_runs = Arc::new(AtomicU64::new(0));
```

Comment:
Honest question: if it's an `AtomicU64`, does it need to be further wrapped in an atomic reference cell?

Reply:
Yes. If it's AtomicU64, it's still an owned type, so it will be moved into the `async move` block. So we need `Arc` for the pattern of sharing owned values across multiple threads/futures.

I also pondered it being a `Arc<u64>`, but that won't work because the value in an `Arc` is immutable--we some integer data structure that allows interior mutability.

It does feel a bit odd to me for just a counter, but perhaps there's a simpler set of data structures I've overlooked.

Reply:
Oh, right, the DB transaction stuff is awkward that way. Good enough!